catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplication.java

```diff
@@ -0,0 +1,249 @@
+import org.codice.ddf.catalog.ui.forms.model.JsonModel.FieldFilter;
+import org.codice.ddf.catalog.ui.forms.model.JsonModel.FormTemplate;
+import org.codice.ddf.catalog.ui.forms.model.JsonTransformVisitor;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.codice.ddf.security.common.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.servlet.SparkApplication;
+
+/** Provides an internal REST interface for working with custom form data for Intrigue. */
+public class SearchFormsApplication implements SparkApplication {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SearchFormsApplication.class);
+
+  private static final Security SECURITY = Security.getInstance();
+
+  private static final ObjectMapper MAPPER =
+      JsonFactory.create(
+          new JsonParserFactory().usePropertyOnly(),
+          new JsonSerializerFactory()
+              .addPropertySerializer(new FilterNodeValueSerializer())
+              .useAnnotations()
+              .includeEmpty()
+              .includeDefaultValues()
+              .setJsonFormatForDates(false));
+
+  private final CatalogFramework catalogFramework;
+
+  private final EndpointUtil util;
+
+  public SearchFormsApplication(CatalogFramework catalogFramework, EndpointUtil util) {
+    this.catalogFramework = catalogFramework;
+    this.util = util;
+  }
+
+  /**
+   * Called via blueprint on initialization. Reads configuration in {@code etc/forms} and
+   * initializes Solr with query and result templates using the {@link Metacard#TITLE} field as a
+   * unique key.
+   */
+  public void setup() {
+    Function<Map<String, Result>, Set<String>> titles =
+        map ->
+            map.entrySet()
+                .stream()
+                .map(Map.Entry::getValue)
+                .map(Result::getMetacard)
+                .map(Metacard::getTitle)
+                .collect(Collectors.toSet());
+
+    Set<String> queryTitles = queryAsAdmin(FormAttributes.Query.TAG, titles);
+    Set<String> resultTitles = queryAsAdmin(FormAttributes.Result.TAG, titles);
+
+    List<Metacard> systemTemplates = config().get();
+    if (systemTemplates.isEmpty()) {
+      return;
+    }
+
+    Stream<Metacard> dedupedTemplateMetacardStream =
+        Stream.concat(
+            systemTemplates
+                .stream()
+                .filter(QueryTemplateMetacardImpl::isQueryTemplateMetacard)
+                .filter(metacard -> !queryTitles.contains(metacard.getTitle())),
+            systemTemplates
+                .stream()
+                .filter(ResultTemplateMetacardImpl::isResultTemplateMetacard)
+                .filter(metacard -> !resultTitles.contains(metacard.getTitle())));
+
+    saveMetacards(dedupedTemplateMetacardStream.collect(Collectors.toList()));
+  }
+
+  /** Spark's API-mandated init (not OSGi related) for registering REST functions. */
+  @Override
+  public void init() {
+    get(
+        "/forms/query",
```

Comment:
❗️ out of the box, without any configuration this call returns a 404 and presents an error on the UI when there are no search forms.  I don't think we should be presenting an error in this case; we should just return an empty JSON response

Reply:
Fixed 👍